onnx_neural_compressor/config.py

```diff
@@ -368,81 +399,104 @@ def get_the_default_value_of_param(config: BaseConfig, param: str) -> Any:
 
         # Get the parameters and their default values
         parameters = signature.parameters
-        return parameters.get(param).default
+        return parameters.get(param).default if parameters.get(param) is not None else None
+
+    @staticmethod
+    def build_tuning_param(config: BaseConfig, param: str):
+        # Create `tuning.TuningParam` for each param
+        # There are two cases:
+        # 1. The param is a string.
+        # 2. The param is a `tuning.TuningParam` instance.
+        if isinstance(param, str):
+            signature = inspect.signature(config.__init__)
+            parameters = signature.parameters
+            default_param = parameters.get(param).default if parameters.get(param) is not None else None
+            tuning_param = TuningParam(name=param, tunable_type=List[type(default_param)])
+        elif isinstance(param, TuningParam):
+            tuning_param = param
+        else:
+            raise ValueError(f"Unsupported param type: {param}")
+        return tuning_param
 
     def expand(self) -> List[BaseConfig]:
         """Expand the config.
 
         case 1
-            {
-                "global": { "weight_bits": [4, 6]}
-            }
-            expand to :
-            1st trial config:
-            {
-                "global": { "weight_bits": 4}
-            }
-            2nd trial config:
-            {
-                "global": { "weight_bits": 6}
-            }
-        case 2
-        # TODO to support the expansion of config with `local`
-        {
-            "global": {
-                "weight_bits": [4, 6]
-            },
-            "local":
-            {
-                "fc1":{
-                    "weight_bits": [6, 8]
-                },
-                "fc2":{
-                    "weight_bits": [4]
-                }
-            }
-
-        } -> ?
+            {"model_params": { "reduce_range": [True, False]}} ->
+            {"model_params": { "reduce_range": True}}, {"model_params": { "reduce_range": False}}
+
+        case 2: iterate op_params first (for this case, Add op only supports per_tensor)
+            {"model_params": { "reduce_range": [True, False]}, "op_params": {"per_channel": [True, False]}} ->
+            {"model_params": { "reduce_range": True}, "op_params": {"per_channel": True}}
+            {"model_params": { "reduce_range": True}, "op_params": {"per_channel": False}}
+            {"model_params": { "reduce_range": False}, "op_params": {"per_channel": True}}
+            {"model_params": { "reduce_range": False}, "op_params": {"per_channel": False}}
+
+            {"model_params": { "reduce_range": [True, False]}, "op_params": {"Conv": {"per_channel": [True, False], , "Add": {"per_channel": [True, False]}}}} ->

```

Comment:
It seems that not all possible combinations are being generated. I suggest adding some comments to explain the generation rule.